snapcraft/internal/pluginhandler.py

```diff
@@ -650,15 +650,34 @@ def _migratable_filesets(fileset, srcdir):
     return snap_files, snap_dirs
 
 
+def _migrate_files_directories(snap_dirs, srcdir, dstdir,
+                               follow_symlinks=False):
+    for directory in snap_dirs:
+        src = os.path.join(srcdir, directory)
+        dst = os.path.join(dstdir, directory)
+        os.makedirs(dst, exist_ok=True)
+        try:
+            shutil.copystat(src, dst, follow_symlinks=follow_symlinks)
+        except FileNotFoundError:
+            logger.info("src not found: {!r}".format(src))
+            pass
+
+
 def _migrate_files(snap_files, snap_dirs, srcdir, dstdir, missing_ok=False,
                    follow_symlinks=False, fixup_func=lambda *args: None):
-    for directory in snap_dirs:
-        os.makedirs(os.path.join(dstdir, directory), exist_ok=True)
 
+    _migrate_files_directories(snap_dirs, srcdir, dstdir,
+                               follow_symlinks=follow_symlinks)
```

Comment:
I don't see a benefit to extracting this into another function now that the conditional snap types are gone. Can we move it back in?